lizard_connector/callbacks.py

```diff
@@ -0,0 +1,73 @@
+# coding=utf-8
+import json
+import time
+import pickle
+
+FILE_BASE = "api_result"
+H5_DATASET_NAME_DATA = "data"
+H5_DATASET_NAME_METADATA = "metadata"
+H5_DATASET_NAMES = (
+    H5_DATASET_NAME_METADATA,
+    H5_DATASET_NAME_DATA,
+)
+
+
+def no_op(*args, **kwargs):
+    pass
+
+
+def save_to_json(result):
+    """
+    Saves a result to json with a timestamp in milliseconds.
+
+    Use with json parser.
+
+    Args:
+        result (list|dict): a json dumpable object to save to file.
+        file_base (str): filename base. Can contain a relative or absolute
+                         path.
+    """
+    filename = "{}_{}.json".format(FILE_BASE, str(int(time.time() * 1000)))
+    with open(filename, 'w') as json_filehandler:
+        json.dump(result, json_filehandler)
+
+
+def save_to_pickle(result):
+    """
+    Pickle a result to file with a timestamp in milliseconds.
+
+    Use with json parser.
+
+    Args:
+        result (list|dict): a python serializable object to save to file.
+        file_base (str): filename base. Can contain a relative or absolute
+                         path.
+    """
+    filename = "{}_{}.p".format(FILE_BASE, str(int(time.time() * 1000)))
+    with open(filename, 'w') as pickle_filehandler:
+        pickle.dump(result, pickle_filehandler)
+
+
+def save_to_hdf5(result):
+    """
+    Saves a result to hdf5 file with a timestamp in milliseconds.
+
+    Use with scientific parser. Requires the h5py library for HDF5.
+
+    Args:
+        result ([pandas.DataFrame|numpy.array]): a hdf5 dumpable object to save
+                                                 to file.
+        file_base (str): filename base. Can contain a relative or absolute
+                         path.
+    """
+    filename = "{}_{}.h5".format(FILE_BASE, str(int(time.time() * 1000)))
+
+    # h5py is only required when using this callback. So we import here.
```

Comment:
Dan is een try/except ImportError eromheen misschien wel zo handig, met een duidelijke foutmelding. Of mis ik context?